src/main.rs

```diff
@@ -127,12 +129,46 @@ fn camera_loop(
                 );
                 return Err(cam_err.err.into());
             }
-            _ => error!(
-                "Error streaming from camera {}, will retry in {}s: {}",
-                camera_config.name,
-                current_backoff.as_secs(),
-                cam_err.err
-            ),
+            neolink::Error::AddrResolutionError => {
+                // Special handling for halting resolution after the resolve timeout period elapses.
```

Comment:
I disagree with the strategy of "stop trying to resolve after a while."  There are many transient conditions that could cause a name resolution failure (DNS server hosed, unpropagated, etc), and we'd like to retry after those. Is there any reason this couldn't simply be removed?

In fact, I'm on the fence about the AuthFailed bailout, for similar reasons.